server/src/collection.rs

```diff
@@ -193,6 +193,29 @@ impl Collection {
         items
     }
 
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let items = self.items.lock().await;
+
+        for item in items.iter() {
+            if !item.is_locked().await {
```

Comment:
This is still very wrong.

Reply:
I have added to avoid emitting PropertiesChanged signal if the item is already locked. This is now moved to the item side.